repos/system_upgrade/el7toel8/actors/systemfacts/libraries/systemfacts.py

```diff
@@ -57,88 +45,99 @@ def anyhasprefix(value, prefixes):
     return False
 
 
-@aslist
-def get_system_users():
+def get_system_users_status():
     ''' Get a list of users from `/etc/passwd` '''
-    for p in pwd.getpwall():
-        yield User(
-            name=p.pw_name,
-            uid=p.pw_uid,
-            gid=p.pw_gid,
-            home=p.pw_dir
-        )
+
+    @aslist
+    def _get_system_users():
+        for p in pwd.getpwall():
+            yield User(
+                name=p.pw_name,
+                uid=p.pw_uid,
+                gid=p.pw_gid,
+                home=p.pw_dir
+            )
+
+    return UsersFacts(users=_get_system_users())
 
 
-@aslist
-def get_system_groups():
+def get_system_groups_status():
     ''' Get a list of groups from `/etc/groups` '''
-    for g in grp.getgrall():
-        yield Group(
-            name=g.gr_name,
-            gid=g.gr_gid,
-            members=g.gr_mem
-        )
 
+    @aslist
+    def _get_system_groups():
+        for g in grp.getgrall():
+            yield Group(
+                name=g.gr_name,
+                gid=g.gr_gid,
+                members=g.gr_mem
+            )
+
+    return GroupsFacts(groups=_get_system_groups())
 
-@aslist
-def get_active_kernel_modules(logger):
+
+def get_active_kernel_modules_status(logger):
     ''' Get a list of active kernel modules '''
-    lines = call(['lsmod'])
-    for l in lines[1:]:
-        name = l.split(' ')[0]
-
-        # Read parameters of the given module as exposed by the
-        # `/sys` VFS, if there are no parameters exposed we just
-        # take the name of the module
-        base_path = '/sys/module/{module}'.format(module=name)
-        parameters_path = os.path.join(base_path, 'parameters')
-        if not os.path.exists(parameters_path):
-            yield ActiveKernelModule(filename=name, parameters=[])
-            continue
-
-        # Use `modinfo` to probe for signature information
-        parameter_dict = {}
-        try:
-            signature = call(['modinfo', '-F', 'signature', name], split=False)
-        except subprocess.CalledProcessError:
-            signature = None
-
-        signature_string = None
-        if signature:
-            # Remove whitspace from the signature string
-            signature_string = re.sub(r"\s+", "", signature, flags=re.UNICODE)
-
-        # Since we're using the `/sys` VFS we need to use `os.listdir()` to get
-        # all the property names and then just read from all the listed paths
-        parameters = sorted(os.listdir(parameters_path))
-        for param in parameters:
+
+    @aslist
+    def _get_active_kernel_modules(logger):
```

Comment:
I am not really sure why you moved those functions within the functions, simply renaming them with an underscore prefix would have been sufficient. This makes this really difficult to review.